<launch>

    <arg name="namespace" default="sam"/>

    <node name="pid_actuator_trim" pkg="sam_basic_controllers" type="pid_actuator_trim" ns="vbs" >
      <param name="topic_from_controller" value="control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="1" />
      <param name="loop_freq" value="50" />
      <remap from="/actuator_input" to="/$(arg namespace)/core/vbs_cmd"/>
    </node>

    <node name="controller" pkg="pid" type="controller" ns="vbs" output="screen" >
      <param name="Kp" value="50.0" />
      <param name="Ki" value="1" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="50.0" />
      <param name="lower_limit" value="-50.0" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/$(arg namespace)/ctrl/depth_feedback" />
      <param name="topic_from_controller" value="control_action" />
      <remap from="setpoint" to="setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>

</launch>
